configuration/pih/scripts/global/zl.js

```diff
@@ -433,4 +433,160 @@ function updateLastCheckboxRequired(containerSelector, transRequiredMsgId, requi
 
   // Add the updateLastCheckboxRequired function to the beforeSubmit array to call it before form submission.
   beforeSubmit.push(updateLastCheckbox);
-}
\ No newline at end of file
+}
+
+patientDossierEditAlert();
+saveSelectedLocation();
+//Creating this function to alert the clinician when He/She wants to modify
+// Patient info which is not in the same location
+function patientDossierEditAlert() {
+  jq(document).ready(function () {
+
+    //Track the previously saved loggeInUserLocation/Clinician from the localStorage
+    let loggedInUserLocation = localStorage.getItem("location");
+    let arraysValue=[]
+ 
+    jq("li.float-left:eq(2)").click(function (e) {
+      // Getting the patient location
+      jq('fieldset.programstatus-fieldset').each(function () {
+       // put all data from the fieldset in a array
+        let firstPTag = jq(this).find('p.programstatus-field:nth-child(2):first');
+        let locations = firstPTag.text().trim();
+        arraysValue.push(locations)
+     
+    });
+    //Remove duplication
+    let uniqueArrays = arraysValue.filter((array, index, self) => {
+  
+      return self.indexOf(array) === index;
+    });
+      //get the location of the patient
+      var patientLocation=uniqueArrays[1];
+
+      if (loggedInUserLocation != patientLocation) {
+        
+        createOverlayAlert(function (){
+          visit.showQuickVisitCreationDialog(0);
+        });
+        return false;
+      } 
+    });
+  });
+}
+//Save teh clinician Location in localStorage
+function saveSelectedLocation() {
+  document.addEventListener("DOMContentLoaded", function () {
+
+    var spanElement = document.getElementById("selected-location");
+    var locationText = spanElement.textContent;
```

Comment:
I thing this is an alternative of what i do @mogoodrich ?

Reply:
@Dutervil  I just noticed a JS console error on this line when working on something else... we should likely make this null-safe, so that it doesn't result in an error on pages without a "selected-location"...  ie, only access "textContent" and set it in local storage if spanElement != null?